pkg/controller/operators/olm/operatorgroup.go

```diff
@@ -100,41 +100,47 @@ func (a *Operator) syncOperatorGroups(obj interface{}) error {
 	logger.WithField("targetNamespaces", targetNamespaces).Debug("updated target namespaces")
 
 	if namespacesChanged(targetNamespaces, op.Status.Namespaces) {
+		logger.Debug("OperatorGroup namespaces change detected")
 		// Labels are only applied to namespaces that are not empty strings.
 		prunedStatusNamespaces := pruneEmptyStrings(op.Status.Namespaces)
 		prunedTargetNamespaces := pruneEmptyStrings(targetNamespaces)
 
-		// Update operatorgroup target namespace selection
-		logger.WithField("targets", targetNamespaces).Debug("namespace change detected")
-		op.Status = v1.OperatorGroupStatus{
-			Namespaces:  targetNamespaces,
-			LastUpdated: a.now(),
-		}
-
-		if _, err = a.client.OperatorsV1().OperatorGroups(op.GetNamespace()).UpdateStatus(op); err != nil && !k8serrors.IsNotFound(err) {
-			logger.WithError(err).Warn("operatorgroup update failed")
-			return err
-		}
-
 		// Remove labels from targetNamespaces that were dropped.
+		logger.Debug("Removing OperatorGroup Labels from dropped namespaces")
 		ogLabel := getOperatorGroupLabel(*op)
 		for _, namespaceName := range setDifference(prunedStatusNamespaces, prunedTargetNamespaces) {
 			err = a.removeNamespaceLabel(namespaceName, ogLabel)
 			if err != nil {
 				logger.WithError(err).Warnf("failed to remove operatorgroup label from namespace %s", namespaceName)
 				return err
 			}
+			logger.Debugf("Removed OperatorGroup label from namespace %s", namespaceName)
 		}
 
 		// Add labels to targetNamespaces that were added.
+		logger.Debug("Adding OperatorGroup Labels to new namespaces")
 		for _, namespaceName := range setDifference(prunedTargetNamespaces, prunedStatusNamespaces) {
```

Comment:
This didn't change in this PR, but there could be a lot of namespaces in a cluster (we scale test with 2000+) which will make syncing an operatorgroup fairly slow (2000*kube api latency)

It would be better to just update the target namespaces when syncing the OperatorGroup, and simply requeue all of the namespaces. 

Syncing a namespace can start by checking if its labels are up to date. If not, we can update them, then exit (requeue again to continue processing the namespace).

Reply:
Will do!